src/Microsoft.Framework.DependencyInjection.Abstractions/Resources.resx

```diff
@@ -133,12 +133,20 @@
     <value>Unable to locate suitable constructor for type '{0}'. Ensure the type is concrete and all parameters are accepted by a constructor.</value>
     <comment>{0} = service type</comment>
   </data>
+  <data name="NoImplementation" xml:space="preserve">
+    <value>Service type '{0}' has been registered without an implementation type.</value>
+    <comment>{0} = service type</comment>
+  </data>
   <data name="NoServiceRegistered" xml:space="preserve">
     <value>No service for type '{0}' has been registered.</value>
     <comment>{0} = service type</comment>
   </data>
-  <data name="TryAddEnumerable_ImplementationTypeMustBeSet" xml:space="preserve">
-    <value>The {0} must have the {1} property set to a non-null value.</value>
-    <comment>{0} = ServiceDescriptor, {1} = ImplementationType</comment>
+  <data name="TryAddIndistinguishableTypeToEnumerable" xml:space="preserve">
+    <value>Unable to add service to Enumerable because the implementation type '{0}' is indistinguishable from the service type '{1}'.</value>
```

Comment:
Any code-like names should also be parameters to the format string to avoid overlocalization.